### DISCUSSION TITLE
Why KingfisherDynamic and DynamicSwiftyJson wrappers?

### DISCUSSION THREAD

**Discussion by gutley** [mozilla-mobile/firefox-ios#12548] [Q&A] [answered]
Just trying to understand why those two wrapper frameworks were needed for the underlying SPM packages they use. Wouldn't SPM have auto-created frameworks for them if they were needed by multiple targets?

**Comment by lmarceau** [comment, marked as answer]
Hello @gutley, KingfisherDynamic was removed in [a recent PR](https://github.com/mozilla-mobile/firefox-ios/pull/12487). But to answer your question, this was needed beginning of this year because of an [SPM bug](https://forums.swift.org/t/swift-packages-in-multiple-targets-results-in-this-will-result-in-duplication-of-library-code-errors/34892). This bug was recently fixed and so we plan on removing the remaining wrappers soon. Thanks!
